test/e2e/specs/specs-wpcom/wp-editor__privacy-password.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * @group gutenberg
+ */
+
+import {
+	DataHelper,
+	GutenbergEditorPage,
+	setupHooks,
+	EditorSettingsSidebarComponent,
+	PostVisibilityOptions,
+	TestAccount,
+	PublishedPostPage,
+	BrowserHelper,
+} from '@automattic/calypso-e2e';
+import { Page } from 'playwright';
+
+describe( DataHelper.createSuiteTitle( `Editor: Post Privacy` ), function () {
+	const accountName = BrowserHelper.targetGutenbergEdge()
+		? 'gutenbergSimpleSiteEdgeUser'
+		: 'simpleSitePersonalPlanUser';
+	const content = DataHelper.getRandomPhrase();
+	const visibility = 'Password';
+	const postPassword = 'car';
+
+	let page: Page;
+	let url: string;
+	let gutenbergEditorPage: GutenbergEditorPage;
+	let editorSettingsSidebarComponent: EditorSettingsSidebarComponent;
+
+	setupHooks( async ( args ) => {
+		page = args.page;
+	} );
+
+	describe( `Create a Password protected post`, function () {
+		beforeAll( async function () {
+			const testAccount = new TestAccount( accountName );
+			await testAccount.authenticate( page );
+		} );
+
+		it( 'Go to the new post page', async function () {
+			gutenbergEditorPage = new GutenbergEditorPage( page );
+			await gutenbergEditorPage.visit( 'post' );
+		} );
+
+		it( 'Enter post title', async function () {
+			gutenbergEditorPage = new GutenbergEditorPage( page );
+			await gutenbergEditorPage.enterTitle(
+				`Post Visibility: ${ visibility } - ${ DataHelper.getTimestamp() }`
+			);
+		} );
+
+		it( 'Enter post content', async function () {
+			await gutenbergEditorPage.enterText( content );
+		} );
+
+		it( `Set post visibility to ${ visibility }`, async function () {
+			const frame = await gutenbergEditorPage.getEditorFrame();
+			editorSettingsSidebarComponent = new EditorSettingsSidebarComponent( frame, page );
+			await editorSettingsSidebarComponent.clickTab( 'Post' );
+			await editorSettingsSidebarComponent.setVisibility( visibility as PostVisibilityOptions );
+			await editorSettingsSidebarComponent.setPostPassword( postPassword );
+		} );
+
+		it( 'Publish post', async function () {
+			url = await gutenbergEditorPage.publish();
+		} );
+	} );
+
+	describe.each( [ accountName, 'defaultUser', 'public' ] )( 'View post as %s', function ( user ) {
```

Comment:
If #59811 merges before this PR, these blocks would be a perfect place to use the on-the-fly new page creation!

Reply:
Yep, now that #59811 has merged I'll be updating the specs here with the new way of doing things.

Reply:
Done.